test/new-e2e/pkg/utils/e2e/client/vm_client.go

```diff
@@ -64,12 +64,54 @@ func NewVMClient(t *testing.T, connection *utils.Connection, osType componentos.
 		privateKeyPassphraseBytes,
 		2*time.Second, 5)
 	return &VMClient{
-		client: client,
-		osType: osType,
-		t:      t,
+		client:     client,
+		connection: connection,
+		osType:     osType,
+		t:          t,
 	}, err
 }
 
+// ReconnectSSH recreate the SSH connection to the VM. Should be used only after VM reboot to restore the SSH connection.
+func (vmClient *VMClient) ReconnectSSH() error {
+	var privateSSHKey, privateKeyPassphraseBytes []byte
```

Comment:
There should a single private function called `connect()` that is called in `New` and `Reconnect`.
Also, generally speaking, rebooting in tests does not sound like the best idea, depending on the use case we may just call `NewVMClient` again.

Reply:
I was thinking about that factorization when I realized I just copy pasted come code, I will make the change.
Regarding the use case, this is something that Windows team will need, because some of their tests need to reboot the host for the change to take effect ( update kernel drivers for example ).
The `Reconnect` function allow the user to recreate the SSH connection without specifying the [Connection](https://github.com/DataDog/test-infra-definitions/blob/main/common/utils/connection.go#L3) that is provided by Pulumi to `NewVMClient` at the creation of the `VMClient`

Reply:
Having the `Connection` object is a non issue with upcoming changes.
Ultimately we should implement auto-reconnect in case of network glitches, SSH daemon restart and these edge reboot cases.

Reply:
Indeed, with the refactoring the `Connection` object info will be present in the `remote.HostOutput` object so it should be easier.
I will create a card to handle automatic reconnection in a follow up PR